lib/CommandClient.php

```diff
@@ -26,22 +32,77 @@ public static function socketPath(string $config) {
     private function send($message): Promise {
         return call(function () use ($message) {
             if (!$this->socket) {
-                $this->socket = yield from $this->establish();
+                if (!$this->socketPromise) {
+                    new Coroutine(self::start($this->path, $this->socket, $this->socketPromise, $this->readDeferreds));
+                }
+
+                yield $this->socketPromise;
+            }
+
+            try {
+                $message = \json_encode($message) . "\n";
+                yield $this->socket->write($message);
+            } catch (ClosedException $e) {
```

Comment:
hmm, you should actually be right as it's anyway a local socket. Will remove that part.